subsys/net/lib/shell/conn.c

```diff
@@ -296,6 +298,31 @@ static int cmd_net_conn(const struct shell *sh, size_t argc, char *argv[])
 	return 0;
 }
 
+static int cmd_net_online_check(const struct shell *sh,
+				size_t argc, char *argv[])
+{
+	bool done;
+
+	done = conn_mgr_trigger_online_connectivity_check();
+	if (!done) {
+		PR("Check %sstarted.\n", "not ");
+		PR_INFO("Set %s to enable %s support.\n",
+		"CONFIG_NET_CONNECTION_MANAGER",
+		"connection manager");
+	} else {
+		PR("Check %sstarted.\n", "");
+	}
+
+	return 0;
+}
+
+SHELL_STATIC_SUBCMD_SET_CREATE(net_cmd_conn,
+	SHELL_CMD(check, NULL,
+		  "'net conn check' trigger online connectivity check.",
+		  cmd_net_online_check),
+	SHELL_SUBCMD_SET_END
+);
 
-SHELL_SUBCMD_ADD((net), conn, NULL, "Print information about network connections.",
+SHELL_SUBCMD_ADD((net), conn, &net_cmd_conn,
+		 "Print information about network connections.",
 		 cmd_net_conn, 1, 0);
```

Comment:
I don't think `lib/shell/conn.c` is actually related to `conn_mgr`

I have been meaning to implement a `conn_mgr` shell that supports manual activation of several of `conn_mgr's` features. That would theoretically be a good place for this

Reply:
Additionally, IMO, the command probably shouldn't be built if connectivity checks are not enabled

Reply:
> Additionally, IMO, the command probably shouldn't be built if connectivity checks are not enabled

I have actually done net-shell differently, so that if the functionality is not there, the command is still present and it tells what to enable to make the command usable. After all the shell is mainly a debugging tool, and it can be difficult to know what commands are available if they are hidden because of some disabled option.

Reply:
> the command is still present and it tells what to enable to make the command usable

My concern there would be a global, non-negotiable extra flash usage. As much as I do appreciate runtime usability, I'd argue the flash savings is worth making the user look up what they need in order to access a given shell.

> After all the shell is mainly a debugging tool

There are definitely some non-debug use-cases for shell to consider. For example, the TLS credentials shell, which allows run-time installation of TLS credentials.

> The purpose of this was to just to way to trigger the check for testing purposes

If that is the case, then I think it should be possible to enable SHELL and online checks without enabling this shell command. I'm happy to do that myself when I add the conn_mgr shell, assuming you are alright with that, since that shell will also be mainly for testing purposes, so all of those commands will similarly need to be optional (and disabled by default).

Reply:
> There are definitely some non-debug use-cases for shell to consider. For example, the TLS credentials shell, which allows run-time installation of TLS credentials.

I slightly disagree here, imho anything requiring manual tweaking in an embedded device is basically a debug tool.

Reply:
> I add the conn_mgr shell, assuming you are alright with that

I am ok with a way to control conn_mgr from the shell, but I would not like to see yet another shell module for this. So if you are doing that, please add the functionality to net-shell.

Reply:
> but I would not like to see yet another shell module for this

Is there a serious disadvantage to it being its own shell? I tend to prefer shell modules be separated similarly to the code they govern. Since conn_mgr is a modular subsystem, IMO it stands to reason that it have its own shell module.

Reply:
My argument for putting connection manager things to net-shell is that conn_mgr is directly related to various network connectivity commands already in net-shell. We could have `net cm <xxx>` for example.
There are different shells for example for wifi and ieee80215 but they are more related to controlling specific network technology instead of general network connectivity.

Reply:
`net cm <xxx>` seems like a reasonable compromise to me. @rlubos Since you're code-owner for the networking subsystem, it seems wise to ask: do you see any potential issues with doing that?

Reply:
Sounds good to me